### ISSUE TITLE
Build status is broken with Gitlab 8.5.8

### ISSUE THREAD

**Issue by vstriz** [jenkinsci/gitlab-plugin#248]
We recently upgraded to Gitlab 8.5.8 (from 8.3.x) and also around same time  upgraded to latest gitlab-plugin 1.1.32 (from 1.1.28). Since then the Builds/build status no longer shows up on Gitlab side.  The build is still triggered in jenkins and it still puts the comment on the MergeRequest, but whether or not it succeeds or fails, gitlab no longer seems to know about the status.

I can try reverting gitlab-plugin to 1.1.28 to see if it's caused by change in plugin or on gitlab side. One of my motivations for upgrading plugin was that I thought it would fix the #186 , but what I'm observing is even bigger problem :(

**Comment by smithrs** [comment]
Just wanted to jump in and say it's working for me. The only issue I'm seeing is that it only shows the most recent build on the page. The minute a new build is started the history of the prior build disappears. Should I open a new bug for that?

To be clear, everything works, including the auto-merge at the end.

EDIT: My bad. I didn't understand how builds and commits map to each other. In my setup developers push to their source branch multiple times while the merge request is open, so it's usually a 1-to-1 mapping.

My advice is to double check the permissions of your Jenkins user, maybe re-generate the private token.

But overall, 8.5.8 looks pretty solid. No show-stoppers so far. Let me know if you want more details about my configuration.

**Comment by vstriz** [comment]
I'm pretty sure it's not configuration, but I could be wrong. It was working for a few months.  I think this is the change on the plugin side as to how to reports back to the Gitlab result of the  build. I'm going to try reverting to the older version(s) to verify....

**Comment by vstriz** [comment]
After some more digging, I was able to resolve the issue. It was configuration. Turns out Jenkins user needs to be a 'Developer' and I had 'Reporter'. I'm pretty sure that used to work before on older Gitlab with Reporter, but it's even documented that it needs to be Developer.

Everything is working fine now, including "Merge when build succeeds". So I'm going to close this issue, as it isn't an issue anymore.  Thanks & Sorry for noise...

Very happy!
